app/javascript/components/components/Header.js

```diff
@@ -1,9 +1,75 @@
 import React, { Component } from 'react'
+import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';
 
 export default class Header extends Component {
+  constructor(props) {
+    super(props);
+    this.toggleNavbar = this.toggleNavbar.bind(this);
+    this.state = {
+      collapsed: true
+    }
+  }
+
+  toggleNavbar() {
+    this.setState({
+      collapsed: !this.state.collapsed
+    });
+  }
+
   render() {
+    const {
+      logged_in,
+      current_user,
+      new_user_route,
+      sign_in_route,
+      sign_out_route
+    } = this.props
+    console.log("logged_in:", logged_in)
+    console.log("current_user:", current_user)
+
     return (
-      <h3>Header</h3>
+      <>
+        <h3>Header</h3>
+        <Navbar color="faded" light>
+          <NavbarBrand href="/" className="mr-auto">Bright Mind</NavbarBrand>
+          <NavbarToggler onClick={this.toggleNavbar} className="mr-2" />
+          <Collapse isOpen={!this.state.collapsed} navbar>
+            <Nav navbar>
+              <NavItem>
+                <a href="/" className="nav-link">Home</a>
+              </NavItem>
+              <NavItem>
+                <a href="/" className="nav-link">About Us</a>
+              </NavItem>
+              { logged_in&&
+                <NavItem>
+                <a href="/postindex" className="nav-link">My Journal</a>
+              </NavItem>
+              }
+              { logged_in &&
+              <NavItem>
+                <a href="/postnew" className="nav-link">Add New Entry</a>
+              </NavItem>
+              }
```

Comment:
Internal navigation should happen using react router dom NavLink with the to='/...' attribute. The main difference here is the forcing of a rerender of the DOM elements by the anchor tag. NavLink does not force a rerender but relies on the react lifecycle to swap mounts.

Reply:
https://v5.reactrouter.com/web/api/NavLink

Reply:
Hey Austin Changes Have been made.
Thank you for the helpful feedback. Would also like to point out that we found out that the NavLink for some reason doesnt work for a tags and I see that you used both A tags AND NavLinks for your example in the apartment app which is what we decided to do as well

Reply:
nav link does not work for the sign in/ sign out / and sign up routes. All it does is changes the url in the browser but does not Rerender the page. So instead we used A tags here. Please Review.